### PR TITLE
#86: Remove dialect-level plug-in mechanism

### LINKED ISSUE #86: Remove dialect-level plug-in mechanism
## Situation

We extracted each dialect into its own repository. There is now a 1 : 1 relationship between adapter and dialect. This makes the current dialect loading superfluous.

Fixes #63.

## Acceptance criteria

* Dialect Registry is removed
* Single dialect is injected

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #86.

### CODE REVIEW

**Review by chiaradiamarcelo — APPROVED** [review]

File: src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapter.java [hunk 1]
```diff
@@ -17,13 +18,13 @@
 /**
  * This class implements main logic for different types of requests a virtual schema JDBC adapter can receive.
  */
-public class JdbcAdapter implements VirtualSchemaAdapter {
+public abstract class AbstractJdbcAdapter implements VirtualSchemaAdapter {
```
[the remark this answers is not in the document]
Comment (chiaradiamarcelo) [reply src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapter.java:21]: My point is, as a user, I don't want to know if the class I'm dealing with is an Interface, a concrete class or an abstract class. All I care is that the I'm dealing with a `JdbcAdapter` and the class provides everything I would expect from a `JdbcAdapter`, no matter how it's implemented. The name should not be picked to communicate implementation, it should communicate abstraction.

[the remark this answers is not in the document]
Comment (chiaradiamarcelo) [reply src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapter.java:21]: My point is, as a user, I don't want to know if the class I'm dealing with is an Interface, a concrete class or an abstract class. All I care is that the I'm dealing with a `JdbcAdapter` and the class provides everything I would expect from a `JdbcAdapter`, no matter how it's implemented. The name should not be picked to communicate implementation, it should communicate abstraction.

[the remark this answers is not in the document]
Comment (redcatbear) [reply src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapter.java:21]: I like to see by the name that a class is abstract.

File: src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapterFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.exasol.logging.VersionCollector;
+
+/**
+ * This class implements a factory for the {@link AbstractJdbcAdapter}.
+ * <p>
+ * Note that this classes derived from this class must be registered in a resource file called
+ * <code>META-INF/services/com.exasol.adapter.AdapterFactory</code> in order for the {@link ServiceLoader} to find it.
+ * </p>
+ */
+public abstract class AbstractJdbcAdapterFactory implements AdapterFactory {
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/main/java/com/exasol/adapter/jdbc/AbstractJdbcAdapterFactory.java:16]: See above.

**Review by chiaradiamarcelo — APPROVED** [review]

**Comment by chiaradiamarcelo** [comment]
Thanks for the fix!

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with one small comment

File: src/test/java/com/exasol/adapter/dialects/derby/DerbyAdapter.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.exasol.adapter.jdbc.AbstractJdbcAdapter;
+import com.exasol.adapter.jdbc.ConnectionFactory;
+
+/**
+ * Virtual Schema Adapter for Apache Derby.
+ * <p>
+ * This adapter is intended for integration tests of the JDBC Virtual Schema adapter only.
+ * </p>
+ */
+public class DerbyAdapter extends AbstractJdbcAdapter implements VirtualSchemaAdapter {
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/test/java/com/exasol/adapter/dialects/derby/DerbyAdapter.java:15]: Fixed.

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**